src/execution/operator/projection/physical_tableinout_function.cpp

```diff
@@ -52,14 +54,30 @@ unique_ptr<GlobalOperatorState> PhysicalTableInOutFunction::GetGlobalOperatorSta
 	return std::move(result);
 }
 
+void PhysicalTableInOutFunction::PrepareOrdinality(DataChunk &chunk, idx_t &ord_index, bool &ord_reset) const {
```

Comment:
Can we deduplicate this logic by making an OrdinalityData struct that encapsulates this

Instead of creating this method on both the PhysicalTableInOut and PhysicalTableScan

Reply:
The new struct OrdinalityData now contains static functions that take care of this logic